src/Evolution/Systems/GeneralizedHarmonic/Constraints.cpp

```diff
@@ -1044,6 +1044,7 @@ void two_index_constraint(
     const Scalar<DataType>& gamma2,
     const tnsr::iaa<DataType, SpatialDim, Frame>&
         three_index_constraint) noexcept {
+  destructive_resize_components(constraint, get_size(get(gamma2)));
```

Comment:
:+1: I think this code  combining `destructive_resize` with the for loop is exactly what would reduce allocations in other place :)